upup/pkg/fi/cloudup/terraform/element.go

```diff
@@ -102,8 +160,50 @@ func ToElement(item interface{}) (element, error) {
 			}
 		}
 		return o, nil
+	case reflect.Slice:
+		if v.Len() == 0 {
+			return nil, nil
+		}
+		elemType := v.Type().Elem()
+		if elemType.Kind() == reflect.Pointer {
+			elemType = elemType.Elem()
+			if elemType == literalType {
+				elements := make([]*terraformWriter.Literal, v.Len())
+				for i := range elements {
+					elem := v.Index(i)
+					elements[i] = elem.Interface().(*terraformWriter.Literal)
+				}
+				return terraformWriter.LiteralListExpression(elements...), nil
+			}
+		}
+		switch elemType.Kind() {
+		case reflect.String:
+			elements := make([]*terraformWriter.Literal, v.Len())
+			for i := range elements {
+				elem := v.Index(i)
+				if elem.Kind() == reflect.Pointer {
+					// TODO can these ever be nil?
+					elem = elem.Elem()
```

Comment:
Agree that panic-ing here probably makes sense.  It would be surprising to have a nil in a list of strings...